SD_Card.h

```diff
@@ -0,0 +1,85 @@
+#include <SD.h>
+#include <SPI.h>
+
+#include "gps.h"
+#include "imu.h"
+
+#pragma once
+
+class SDCard {
+  private:
+    File m_sdCardDataFile;
+    static const int m_chipSelect = 4; // this is not the final value - SDCARD_SS_PIN is the correct one for the MKR Zero
+    static const String m_fileName = "rocketry_data";
+  public:
+    void writeToCSV( // this is equivalent 
+      const GPS::Coordinates& coords,
```

Comment:
This is partly a design-level question that warrants more discussion, so I'll ping @DarthSethious1 as well.

The GPS updates far less frequently than the other sensors. For the radio, to keep the frame shape consistent, we accepted that sometimes the packet will have outdated GPS data. However, we don't have that restriction here: we can leave GPS values blank instead of putting older data. Which do we want to do?

If we decide to put older data so that every row has something, nothing needs to change here. If we decide to put blanks where we don't have up-to-date data, that would necessitate one of two changes:

- Add a second version of the function without this parameter
- Pass it by pointer instead (`const GPS::Coordinates* coordsPtr`) and pass `nullptr` when no new data is available

I don't have a preference for which route we take.

Reply:
I think keeping this consistent between radio and SD card as often as possible is good. I'm fine with us writing old GPS values to SD Card and only updating it a fraction of the frames.